src/annotator/config/config.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var settings = require('./settings');
+var sharedSettings = require('../../shared/settings');
+
+/**
+ * Reads the Hypothesis configuration from the environment.
+ *
+ * @param {Window} window_ - The Window object to read config from.
+ */
+function configFrom(window_) {
+  var config = {};
+
+  config.app = settings.iFrameSrc(window_.document);
+
+  Object.assign(config, sharedSettings.jsonConfigsFrom(window_.document));
+  Object.assign(config, settings.configFuncSettingsFrom(window_));
+
+  // Convert legacy keys/values in config to corresponding current
+  // configuration.
+  if (typeof config.showHighlights === 'boolean') {
+    config.showHighlights = config.showHighlights ? 'always' : 'never';
+  }
```

Comment:
In the future there will be a separate `settings.showHighlights()` function for returning the value of the `config.showHighlights` setting, and this transformation code can go into that function. (And in general any validation etc that we want to do for any config setting can go into that setting's own function.)